pkg/apis/core/v1beta1/defaults_shoot.go

```diff
@@ -295,6 +296,21 @@ func SetDefaults_Worker(obj *Worker) {
 	if obj.UpdateStrategy == nil {
 		obj.UpdateStrategy = ptr.To(AutoRollingUpdate)
 	}
+
+	if obj.UpdateStrategy != nil && (*obj.UpdateStrategy == AutoInPlaceUpdate || *obj.UpdateStrategy == ManualInPlaceUpdate) {
+		if obj.MachineControllerManagerSettings == nil {
+			obj.MachineControllerManagerSettings = &MachineControllerManagerSettings{}
+		}
+		if obj.MachineControllerManagerSettings.DisableHealthTimeout == nil {
+			obj.MachineControllerManagerSettings.DisableHealthTimeout = ptr.To(true)
+		}
+
+		// In case of manual in-place update, we set the MaxSurge to 0 and MaxUnavailable to 1.
+		if *obj.UpdateStrategy == ManualInPlaceUpdate {
+			obj.MaxSurge = ptr.To(intstr.FromInt32(0))
+			obj.MaxUnavailable = ptr.To(intstr.FromInt32(1))
+		}
```

Comment:
Can you document this somewhere? I guess it could be surprising for users that want to customize these settings